ls1intum/Artemis
# `Assessment`: Allow instructors to export exam scores to an Excel file

### Checklist
#### General


#### Client

### Motivation and Context


Currently, exporting to Excel is only supported for course scores, but not for exam scores.


### Description

This change adds the Excel Export functionality (already introduced for course scores in #4941) for exam scores.
In addition, custom CSV export options (already introduced for course scores in #4869) can now be specified when exporting exam scores.
All row builder source files are moved from `course-scores` to the `shared/export folder` and are renamed accordingly, too.

### Steps for Testing

- 1 Course
- min. 1 Student
- 1 Exam, on which the student(s) participated

1. Log in to Artemis
2. Navigate to Course Administration, select the course, Exam, then Scores
3. Click the `Export` 
4. A modal opens, where you can select the file type used for the export (Excel<default> or CSV).
5. Select Excel (which should be selected initially) and click `Export as Excel`
6. Open the downloaded Excel file and make sure the rows and their values are equivalent to the ones displayed in Artemis
7. In addition, also verify that the export to a CSV file still works as expected (the custom CSV export options are incorporated).
8. Please also test the export for course scores as well (according to the test instructions in #4941)

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| exam-scores.component.ts | 83.11% | 96.4% |
| csv-export-row-builder.ts | 100% | 100% |
| excel-export-row-builder.ts | 100% | 100% |
| export-row-builder.ts | 96% | 100% |



### Screenshots


![Exam_Excel_Export](https://user-images.githubusercontent.com/9954674/167667737-028acfb9-bc07-47ce-a716-8f0a404a3c13.gif)
